roles/foreman_provisioning/tasks/main.yml

```diff
@@ -145,82 +144,78 @@
+  with_items:
+    - CentOS mirror
+    - CoreOS mirror
+    - Debian mirror
+    - Fedora Atomic mirror
+    - Fedora mirror
+    - FreeBSD mirror
+    - OpenSUSE mirror
+    - Ubuntu mirror
   shell: >
-    {{ foreman_provisioning_hammer }} --output json os info --name "CentOS 7" ||
-    {{ foreman_provisioning_hammer }} --output json os info --title "CentOS 7"
-  register: foreman_provisioning_centos73_json
-  ignore_errors: True
+    {{ foreman_provisioning_hammer }} medium update --name "{{ item }}" --organization-titles "{{ foreman_provisioning_organization }}" --location-titles "{{ foreman_provisioning_location }}"
 
-- name: 'get CentOS 7 info'
-  set_fact:
-    foreman_provisioning_centos73: "{{ foreman_provisioning_centos73_json.stdout|from_json }}"
 
-- name: 'find kickstart templates'
-  shell: >
-    {{ foreman_provisioning_hammer }} --output=json template list
-    --search 'name ~ "Kickstart default"'
-  register: kickstart_templates_json
+- name: 'Setup CentOS 7 provisioning'
+  import_tasks: configure_centos_7.yml
 
-- name: 'set kickstart templates'
-  set_fact:
-    kickstart_templates: "{{ kickstart_templates_json.stdout|from_json }}"
+- name: 'Setup Fedora 27 provisioning'
+  import_tasks: configure_fedora_27.yml
 
-- name: 'associate kickstart templates to CentOS 7'
-  shell: >
-    {{ foreman_provisioning_hammer }} template add-operatingsystem --id {{ item.Id }} --operatingsystem 'CentOS 7'
-  with_items: "{{ kickstart_templates }}"
+- name: 'Setup Debian 9.3 provisioning'
+  import_tasks: configure_debian_9_3.yml
 
-- name: 'set default templates for CentOS 7'
-  shell: >
-    {{ foreman_provisioning_hammer }} os set-default-template --id {{ foreman_provisioning_centos73.Id }} --config-template-id {{ item.Id }}
-  with_items: "{{ kickstart_templates }}"
+- name: 'Setup Ubuntu 17.10 provisioning'
+  import_tasks: configure_ubuntu_17_10.yml
 
+# TODO verify - should not be needed, DNS should work just fine
 - name: 'set unattended_url'
   shell: >
     {{ foreman_provisioning_hammer }} settings set --name unattended_url --value 'http://{{ foreman_provisioning_ip_address }}'
 
-- name: 'find hostgroup Forklift Base'
+- name: 'find hostgroup Base'
   shell: >
-    {{ foreman_provisioning_hammer }} hostgroup info --name 'Forklift Base'
+    {{ foreman_provisioning_hammer }} hostgroup info --name 'Base'
   register: foreman_provisioning_hostgroup_base
   ignore_errors: True
 
-- name: 'create hostgroup Forklift Base'
+# TODO can't set environment http://projects.theforeman.org/issues/22517
+# when it's fixed add this:
+# --environment production
+# TODO compute-profile can't be specified by name until http://projects.theforeman.org/issues/21580/ so we hardcode 1
+- name: 'create hostgroup Base'
   shell: >
     {{ foreman_provisioning_hammer }} hostgroup create
-    --name 'Forklift Base'
+    --name 'Base'
     --architecture x86_64
     --domain {{ foreman_provisioning_domain }}
-    --environment production
```

Comment:
Why removing this? It means I can't use the host group directly and need to set the puppet environment manually later.

Reply:
see the TODO above, can't set environment, it seems to be fixed now so removing it